src/domains/ecommerce-app/api/ecommerce-checkout/v3/_payment_request_policy.xml.tpl

```diff
@@ -0,0 +1,27 @@
+<policies>
+  <inbound>
+    <!-- pass rptId value into header START -->
+    <set-header name="x-rpt-id" exists-action="delete" />
+    <set-variable name="rptId" value="@{
+      return context.Request.MatchedParameters["rpt_id"];
+    }" />
+    <choose>
```

Comment:
is it possible to move this logic in base policy? I think it is safe for all API because there is when condition

Reply:
Should we have an rptId param in all request?

Reply:
`rptId` is not present in all APIs, but with the `when` condition it is set only when present. This way, we avoid having a dedicated policy for a specific API, but we should consider the trade-off. It's an extra operation, although very light, but applied to all APIs.

Reply:
So, it's a suggestion, we might not do it after all :-)